types/recoil/lib/core/atom.d.ts

```diff
@@ -0,0 +1,11 @@
+import { NodeKey } from './state';
+import { RecoilValue, RecoilState } from './recoilValue';
+
+export type AtomOptions<T> = Readonly<{
```

Comment:
Might be worth considering using an interface instead of a type. The advantage of using an interface is that the name replaces the literal type in user facing messages e.g. errors. 

With type:

```
Argument of type '{ key: string; dangerouslyAllowMutability: false; }' is not assignable to parameter of type 'Readonly<{ key: unknown; default: unknown; dangerouslyAllowMutability?: boolean | undefined; }>. Property 'default' is missing.'.
```

With interface:

```
Argument of type '{ key: string; dangerouslyAllowMutability: false; }' is not assignable to parameter of type 'AtomOptions2<unknown>. Property 'default' is missing.'
```

https://www.typescriptlang.org/play?target=1#code/KYDwDg9gTgLgBDAnmYcCCMIFsDyYYCWEAdgM4A8AKgHxwC8cASsAIYAmJANouQN4CwAKDgi4Aa2CIAXHACuxMcQgB3YgG4houG2AAzFrM4wZABSjYCpYFVoAfOJQ3DRbFsQDmwc7NLc0nThUAWVkYFgAjAk4CJAB+GXCICE5WdSEAX2onIR0AY04WKFRdeVzCEgQWCTxyshsACggZDGwaojqaAEoZADcIAjZswTDq-HbSet4JaTgAcgBGWYAabTdPb19Ef0DlELDI6KQZfU4rdM6hIVBIWDgCYhgvfVzUFtwxklIAJhs4UEfiGxSExWBxiNw+JpRNMZPJFCo0s4RDp9IZjHAzBYrL97I4oci1l4ID4-AFgqEIlEYoh4nBEslUk5Mrx0pdBHkCkU4LlPvAIF9mph3rVvuRSDAoPd3FkcsB8oViqVapVRiKfjRGoLWh8yOrqN04H0BkMRsA2p8vpMYXNFitXB4iSStmTdhSDtTjixTsBzkA

Reply:
good idea! Interfaces will also make the types a little friendlier to extensibility in user-land, so I'll try making the type exports interface-based wherever possible